### PR TITLE
Improve test coverage for VectorsCombiner and make vector aggregator efficient 

### PR DESCRIPTION
**Pull request by tovbinm**
**Related issues**
1. `UnionVector` monoid aggregator is not efficient for sparse vectors.
2. `VectorsCombiner` stage has incomplete functional test coverage.

**Describe the proposed solution**
1. Moved `VectorsCombiner.combine` method to `RichVector` object
2. Added `OpEstimatorSpec` base class to `VectorsCombinerTest` to improve the functional test coverage
3. Added `+`, `-`, `dot`,`combine` methods to `OPVector` + test `OPVectorTest`
4. Added `combine` method to `RichVector` + tests
5. Renamed `UnionVector` monoid aggregator to `CombineVector` and made use of `RichVector.combine` at reduve
6. Bonus: added `SumVector` monoid aggregator

**Describe alternatives you've considered**
I could relocate `VectorsCombiner.combine` as-is to `utils` subproject, but it felt valuable enough to add `+`, `-`,`dot` and `combine` to `OPVector` as well.

### CODE REVIEW

File: features/src/main/scala/com/salesforce/op/features/types/OPVector.scala [hunk 1]
```diff
@@ -39,8 +40,37 @@ import org.apache.spark.ml.linalg._
...
+  def +(that: OPVector): OPVector = (value + that.value).toOPVector
+
+  /**
+   * Subtract vectors
+   *
+   * @param that another vector
+   * @throws IllegalArgumentException if the vectors have different sizes
+   * @return vector subtraction
+   */
+  def -(that: OPVector): OPVector = (value - that.value).toOPVector
```
Comment (leahmcguire) [inline features/src/main/scala/com/salesforce/op/features/types/OPVector.scala]: as long as you are adding addition and subtraction why not dot product and cross product?

File: features/src/main/scala/com/salesforce/op/utils/spark/RichVector.scala [hunk 2]
```diff
@@ -64,6 +66,16 @@ object RichVector {
...
 
+    /**
+     * Combine multiple vectors into one
+     *
+     * @param that  another vector
+     * @param other other vectors
+     * @return result vector
+     */
+    def combine(that: Vector, other: Vector*): Vector =
+      com.salesforce.op.utils.spark.RichVector.combine(v +: that +: other)
```
Comment (leahmcguire) [inline features/src/main/scala/com/salesforce/op/utils/spark/RichVector.scala]: do you need the full import here?

File: features/src/test/scala/com/salesforce/op/features/types/OPVectorTest.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    for {(v1, v2) <- vectors.zip(vectors)} {
+      (v1 + v2) shouldBe (v1.value + v2.value).toOPVector
+    }
+  }
+
+  it should "'-' subtract correctly" in {
+    for {(v1, v2) <- vectors.zip(vectors)} {
+      (v1 - v2) shouldBe (v1.value - v2.value).toOPVector
+    }
+  }
```
Comment (leahmcguire) [inline features/src/test/scala/com/salesforce/op/features/types/OPVectorTest.scala]: maybe tests for the errors when the vectors have different lengths?

File: features/src/main/scala/com/salesforce/op/utils/spark/RichVector.scala (hunk 2 above)
Comment (tovbinm) [reply features/src/main/scala/com/salesforce/op/utils/spark/RichVector.scala]: this is to distinguish the object from the implicit class.

**Comment by tovbinm** [comment]
@leahmcguire done

**Pull request merged**